as.c

```diff
@@ -251,16 +269,10 @@ static void assemble_line(vm_env *env, char *line)
     free(line_backup);
 }
 
-void assemble_from_file(vm_env *env, const char *filename)
+void assemble_from_file(vm_env *env, FILE *fp)
```

Comment:
1. void assemble_from_file(vm_env *env, FILE *fp);
2. int write_to_elf_file(vm_env *env, int fd);
3. int load_from_elf_file(vm_env *env, int fd);

In all three APIs, if 1 uses string but the other two uses file descriptor, they would not be on par, like below:

1. void assemble_from_file(vm_env *env, char *filename);
2. int write_to_elf_file(vm_env *env, int fd);
3. int load_from_elf_file(vm_env *env, int fd);

If they were to be on par, using 'char *filename', like below:

1. void assemble_from_file(vm_env *env, char *filename);
2. int write_to_elf_file(vm_env *env, char *filename);
3. int load_from_elf_file(vm_env *env, char *filename);

It means the detection of availability of target file will be implemented in these functions.